examples/7GUIs/circle_drawer.rs

```diff
@@ -0,0 +1,358 @@
+use vizia::prelude::*;
+use vizia_core::vg::{Paint, Path};
+
+const STYLE: &str = r#"
+    :root {
+        child-space: 10px;
+    }
+
+    .header {
+        left: 1s;
+        right: 1s;
+        col-between: 5px;
+    }
+
+    circle-drawer {
+        row-between: 10px;
+    }
+
+    circle-drawer-canvas {
+        border-color: black;
+        border-width: 2px;
+    }
+
+    .dialog-box {
+        child-space: 1s;
+        width: 460px;
+        height: 100px;
+        top: 1s;
+        left: 1s;
+        right: 1s;
+        bottom: 40px;
+        background-color: rgba(255, 255, 255, 0.7);
+        border-color: black;
+        border-width: 1px;
+        border-radius: 10%;
+    }
+
+    .dialog-box vstack {
+        child-space: 1s;
+        row-between: 15px;
+    }
+
+    .dialog-box vstack slider {
+        width: 400px;
+    }
+"#;
+
+fn distance(x1: f32, y1: f32, x2: f32, y2: f32) -> f32 {
+    f32::sqrt(f32::powf(x1 - x2, 2.0) + f32::powf(y1 - y2, 2.0))
+}
+
+#[derive(Clone, Copy, Data)]
+struct Circle {
+    x: f32,
+    y: f32,
+    r: f32,
+}
+
+#[derive(Clone, Default, Data, Lens)]
+struct CircleData {
+    /// circles
+    circles: Vec<Circle>,
+    /// index of current selected circle
+    selected: Option<usize>,
+}
+
+enum UndoRedoAction {
+    Circle(Circle),
+    RadiusChange(usize, f32),
+}
+
+#[derive(Lens)]
+struct CircleDrawerData {
+    circles_data: CircleData,
+    /// Undo redo
+    undo_list: Vec<UndoRedoAction>,
+    redo_list: Vec<UndoRedoAction>,
+    radius_before: f32,
+    /// is right click menu open
+    menu_open: bool,
+    menu_posx: Units,
+    menu_posy: Units,
+    /// is dialog box open
+    dialog_open: bool,
+}
+
+impl Default for CircleDrawerData {
+    fn default() -> Self {
+        Self {
+            circles_data: CircleData::default(),
+            undo_list: Vec::new(),
+            redo_list: Vec::new(),
+            radius_before: 0.0,
+            menu_open: false,
+            menu_posx: Default::default(),
+            menu_posy: Default::default(),
+            dialog_open: false,
+        }
+    }
+}
+
+enum CircleDrawerEvent {
+    AddCircle(f32, f32),
+    TrySelectCircle(f32, f32),
+    ChangeRadius(f32),
+    Undo,
+    Redo,
+    ToggleRightMenu,
+    ToggleDialog,
+}
+
+impl CircleDrawerData {
+    fn update_selected(&mut self, x: f32, y: f32) {
+        self.circles_data.selected =
+            self.circles_data.circles.iter().position(|c| distance(c.x, c.y, x, y) < c.r);
+    }
```

Comment:
This will find the first circle where the mouse is within the radius, but what about overlapping circles? In this case it might find the wrong circle as you'd expect it to select the circle 'on top', which would be *after* the first one found.

Reply:
One easy solution would be to add a `rev()` to the iterator so you're searching front to back.